modules/custom/d_p/src/Plugin/Field/FieldWidget/SettingsWidget.php

```diff
@@ -44,13 +45,30 @@ class SettingsWidget extends WidgetBase {
   const HEADING_TYPE_SETTING_NAME = 'heading_type';
 
   /**
-   * Get configuration options for fields in paragraph settings.
+   * The paragraph setting plugin manager.
+   *
+   * @var \Drupal\d_p\ParagraphSettingPluginManagerInterface
    */
-  private static function getConfigOptions() {
-    /** @var \Drupal\d_p\ParagraphSettingPluginManagerInterface $pluginManager */
-    $pluginManager = \Drupal::service('d_p.paragraph_settings.plugin.manager');
+  protected $paragraphSettingsManager;
+
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
```

Comment:
this method is missing a comment - not that I require it.. but maybe you do ? ;)